ru.bmstu.rk9.rao/src/ru/bmstu/rk9/rao/jvmmodel/VarConstCompiler.xtend

```diff
@@ -0,0 +1,109 @@
+package ru.bmstu.rk9.rao.jvmmodel
+
+import org.eclipse.xtext.common.types.JvmDeclaredType
+import org.eclipse.xtext.common.types.JvmVisibility
+import org.eclipse.xtext.naming.QualifiedName
+import org.eclipse.xtext.xbase.jvmmodel.JvmTypeReferenceBuilder
+import org.eclipse.xtext.xbase.jvmmodel.JvmTypesBuilder
+import ru.bmstu.rk9.rao.rao.VarConst
+import java.util.ArrayList
+import org.eclipse.xtext.common.types.impl.JvmFormalParameterImplCustom
+import java.util.Arrays
+import java.util.HashMap
+
+class VarConstCompiler extends RaoEntityCompiler {
+	def static asClass(VarConst varconst, JvmTypesBuilder jvmTypesBuilder, JvmTypeReferenceBuilder typeReferenceBuilder,
+		JvmDeclaredType it, boolean isPreIndexingPhase) {
+		initializeCurrent(jvmTypesBuilder, typeReferenceBuilder);
+		
+		val vcQualifiedName = QualifiedName.create(qualifiedName, varconst.name)
+		
+		return varconst.toClass(vcQualifiedName) [
+			static = true
+			superTypes += typeRef(ru.bmstu.rk9.rao.lib.varconst.VarConst)
+
+			members += varconst.toConstructor [
+				visibility = JvmVisibility.PRIVATE
+				parameters += varconst.start.toParameter("start", typeRef(Double))
+				parameters += varconst.stop.toParameter("stop", typeRef(Double))
+				parameters += varconst.step.toParameter("step", typeRef(Double))
+				if (varconst.lambda !== null)
+					parameters += varconst.lambda.toParameter("lambda", typeRef(ru.bmstu.rk9.rao.lib.lambdaexpression.LambdaExpression))
+				body = '''
+					«FOR param : parameters»this.«param.name» = «param.name»;
+					«ENDFOR»
+				'''
+			]
+			
+			members += varconst.start.toField("start", typeRef(Double))
+			members += varconst.stop.toField("stop", typeRef(Double))
+			members += varconst.step.toField("step", typeRef(Double))
+			if (varconst.lambda !== null)
+				members += varconst.lambda.toField("lambda", typeRef(ru.bmstu.rk9.rao.lib.lambdaexpression.LambdaExpression))
+			
+			members += varconst.toMethod("getName", typeRef(String)) [
+				visibility = JvmVisibility.PUBLIC
+				final = true
+				body = '''
+					return "«vcQualifiedName»";
+				'''
+			]
+			
+			members += varconst.toMethod("checkValue", typeRef(Boolean)) [
```

Comment:
Here new checkValue method (old version is a checkLambda now). I just call it in new method